source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -1439,10 +1479,180 @@ TEST_F(NiDAQmxDriverApiTests, ChannelWithDoneEventRegisteredTwice_RunCompleteFin
   read_analog_f64(FINITE_SAMPLE_COUNT, FINITE_SAMPLE_COUNT);
   RegisterDoneEventResponse response1, response2;
   read_stream(reader_context1, *reader1, response1);
-  read_stream(reader_context2, *reader2, response2);
 
   EXPECT_EQ(DAQMX_SUCCESS, response1.status());
-  EXPECT_EQ(DONE_EVENT_ALREADY_REGISTERED_ERROR, response2.status());
+  EXPECT_THROW({
+    try {
+      read_stream(reader_context2, *reader2, response2);
+    }
+    catch (const client::grpc_driver_error& ex) {
+      expect_driver_error(ex, DONE_EVENT_ALREADY_REGISTERED_ERROR);
+      throw;
+    }
+  }, client::grpc_driver_error);
+}
+
+TEST_F(NiDAQmxDriverApiTests, EveryNSamplesEventRegisteredWithWrongEventType_ReadStream_NotSupportedByDeviceErrorReceived)
+{
+  const auto N_SAMPLES = 10UL;
+  create_ai_voltage_chan(0.0, 1.0);
+  ::grpc::ClientContext reader_context;
+  auto reader = register_every_n_samples_event(reader_context, N_SAMPLES, EveryNSamplesEventType::EVERY_N_SAMPLES_EVENT_TYPE_TRANSFERRED_FROM_BUFFER);
+
+  RegisterEveryNSamplesEventResponse response;
+  EXPECT_THROW({
+    try {
+      read_stream(reader_context, *reader, response);
+    }
+    catch (const client::grpc_driver_error& ex) {
+      expect_driver_error(ex, EVERY_N_SAMPS_TRANSFERRED_FROM_BUFFER_EVENT_NOT_SUPPORTED_BY_DEVICE_ERROR);
+      throw;
+    }
+  }, client::grpc_driver_error);
+}
+
+TEST_F(NiDAQmxDriverApiTests, EveryNSamplesEventRegisteredWithWrongEventType_Finish_NotSupportedByDeviceErrorReceived)
+{
+  const auto N_SAMPLES = 10UL;
+  create_ai_voltage_chan(0.0, 1.0);
+  ::grpc::ClientContext reader_context;
+  auto reader = register_every_n_samples_event(reader_context, N_SAMPLES, EveryNSamplesEventType::EVERY_N_SAMPLES_EVENT_TYPE_TRANSFERRED_FROM_BUFFER);
+
+  EXPECT_THROW({
+    try {
+      auto status = reader->Finish();
+      client::raise_if_error(status, reader_context);
+    }
+    catch (const client::grpc_driver_error& ex) {
+      expect_driver_error(ex, EVERY_N_SAMPS_TRANSFERRED_FROM_BUFFER_EVENT_NOT_SUPPORTED_BY_DEVICE_ERROR);
+      throw;
+    }
+  }, client::grpc_driver_error);
+}
+
+TEST_F(NiDAQmxDriverApiTests, EveryNSamplesEventRegisteredWithWrongEventType_Cancel_NoErrors)
+{
+  const auto N_SAMPLES = 10UL;
+  create_ai_voltage_chan(0.0, 1.0);
+  ::grpc::ClientContext reader_context;
+  auto reader = register_every_n_samples_event(reader_context, N_SAMPLES, EveryNSamplesEventType::EVERY_N_SAMPLES_EVENT_TYPE_TRANSFERRED_FROM_BUFFER);
+
+  reader_context.TryCancel();
+}
+
+TEST_F(NiDAQmxDriverApiTests, AsyncEveryNSamplesEventRegistered_OnEventReadAnalogF64_EventAndReadResponsesAreReceived)
```

Comment:
```
[ RUN      ] NiDAQmxDriverApiTests.AsyncEveryNSamplesEventRegistered_OnEventReadAnalogF64_EventAndReadResponsesAreReceived
/opt/actions-runner/_work/_temp/8[92](https://github.com/ni/grpc-device/actions/runs/5047577024/jobs/9064618937#step:5:93)d9ccb-3fdf-43e6-b8c3-b413564be71f.sh: line 1:  3144 Segmentation fault      (core dumped) ./SystemTestsRunner --gtest_filter=-NiRFSA*:NiRFSG*:NiDCPower*CalSelfCalibrate*
Error: Process completed with exit code 139.
```

Reply:
I set up a Linux VM to debug this, and the main branch crashes without the change from this PR: https://github.com/ni/grpc-device/issues/936

Reply:
With the `make_scope_exit` fix, both `run_win_system_tests` and `run_linux_system_tests` are now passing.